### PR TITLE
Establish API base

### LINKED ISSUE #1: Creation of the Pre-launch Landing page API
## What needs to be delivered
The backend API for the landing page that collects emails and sends them to a postgres database.

### How should it be done?
The email form should be validated server side for a valid email, then posted to the database with an unverified status. Then a verification email via Sendgrid should be sent to said email and change the record to verified once verified. Upon successful verification a webhook should be fired for an internal activity feed.

### Technical Considerations
Already sorted, issue created after API was finished.

### Specs or Mocks 
See https://github.com/freelyis/placeholder/issues/2

### Addition context
See https://github.com/freelyis/placeholder/issues/2

### PR DESCRIPTION
**Pull request by caf203**
Resolves #1 

Note that the SendGrid message is hardcoded and unstyled. We need to get someone to create the template to be sent to users when they sign up for the news letter.

### CODE REVIEW

**Comment by pizzafox** [comment]
`cURL` example for both endpoints:

```sh
curl --request POST \
  --url https://freely.is/api/submitinterest \
  --header 'content-type: application/json' \
  --data '{
	"email": "jonah@jonah.pw"
}'
```

```sh
curl --request GET \
  --url https://freely.is/api/verifyinterest/d603458f-5287-4076-b773-93a52f6d6d33
```

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/node_modules/
```
Comment (MattIPv4) [inline .eslintignore]: Newline at end of file?

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (MattIPv4) [inline .eslintrc.json]: I thought the intention was to use a centralised eslint config that each repo then extended?

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        }],
+
+        "prefer-destructuring": ["error", {
+            "array": false,
+            "object": true
+        }, {
+            "enforceForRenamedProperties": true
+        }]
+    }
+}
```
Comment (MattIPv4) [inline .eslintrc.json]: Newline at end of file?

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+node_modules
+.env
```
Comment (MattIPv4) [inline .gitignore]: Newline at end of file?

File: .eslintignore (hunk 1 above)
Comment (MattIPv4) [inline .eslintignore]: Can this just be `node_modules` for consistency with the gitignore? I believe eslint should still resolve that to the dir correctly?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,56 @@
...
+1. Clone
+
+```bash
+git clone https://github.com/freelyis/placeholder-api
+```
+
+2. Install dependencies
+
+```bash
+npm i
```
Comment (MattIPv4) [inline README.md]: A package-lock.json file is provided, this should be `npm ci` really.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,56 @@
...
+## Getting Started
+
+These instructions will get you a copy of the project up and running on your local machine for development and testing purposes. See deployment for notes on how to deploy the project on a live system.
+
+### Prerequisites
+
+What things you need to install the software and how to install them
+
+```
+NodeJS (12 preferrably)
```
Comment (MattIPv4) [inline README.md]: Why not specify a node version with an `.nvmrc` file?

File: src/sendgrid.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+require("dotenv").config();
+
+mailer.setApiKey(process.env.SENDGRID_API_KEY);
+
+exports.sendSuccessEmail = async (email) => {
+    await mailer.send({
+        to: email,
+        from: "toucan@freely.is",
+        subject: "Newsletter Sign Up",
+        text: "Toucan", // this does not render when html is populated
```
Comment (MattIPv4) [inline src/sendgrid.js]: Should this not be fallback text for if a email client cannot use the html version?

File: src/sendgrid.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        html: "Hello User! <br />Thank you for your interest! We will send updates to you now. <br /><br />Cheers,<br />Toucan",
+    });
+};
+
+exports.sendVerification = async (email, uuid) => {
+    await mailer.send({
+        to: email,
+        from: "toucan@freely.is",
+        subject: "Newsletter Sign Up",
+        text: "Toucan", // this does not render when html is populated
```
Comment (MattIPv4) [inline src/sendgrid.js]: See other comment, fallback text?

File: .eslintrc.json (hunk 2 above)
Comment (MattIPv4) [reply .eslintrc.json]: Ignore, this will happen for prod, not this.

File: src/sendgrid.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+        html: `Hello User! <br />Thank you for your interest in our newsletter. Please click <a href=http://localhost:8080/api/verifyinterest/${uuid}>this link</a> to verify that you\'re not a bot. <br /><br />Cheers,<br />Toucan`, // eslint-disable-line no-useless-escape
```
[the remark this answers is not in the document]
Comment (caf203) [reply src/sendgrid.js]: This is temporary as we do not have a domain. I'll implement this though.

File: .eslintignore (hunk 1 above)
Comment (caf203) [reply .eslintignore]: This was automatically done from vscode. I'll switch it now.

File: README.md (hunk 5 above)
Comment (caf203) [reply README.md]: Done

File: README.md (hunk 6 above)
Comment (caf203) [reply README.md]: done

File: src/sendgrid.js (hunk 7 above)
Comment (caf203) [reply src/sendgrid.js]: being worked on right now in the discord

**Pull request merged**